taker-electron/src/main/main.ts

```diff
@@ -74,6 +97,16 @@ const createWindow = async () => {
         await installExtensions();
     }
 
+    if (!tray) { // if tray hasn't been created already.
```

Comment:
I like it. I kept the tray icon though even if the window is there.

Reply:
at least on macOS, this is the expected behaviour and that's what the apps do. I would recommend to follow established UX conventions rather than making things more clever (but unexpected).
can't say about windows, as I haven't used that one in ages.
PS on macOS, there is typically a switch that removes app from the dock altogether, relying only on the tray icon. We might want to implement that someday.